core/src/main/kotlin/fr/sncf/osrd/api/api_v2/RequirementsParser.kt

```diff
@@ -66,3 +79,89 @@ fun parseRoutingRequirements(
     }
     return res
 }
+
+fun parseWorkSchedulesRequest(
+    infra: RawSignalingInfra,
+    workSchedulesRequest: WorkSchedulesRequest,
+    startTime: ZonedDateTime
+): Collection<Requirements> {
+    val delta = TimeDelta(between(startTime, workSchedulesRequest.startTime).toMillis())
+    return convertWorkSchedules(infra, workSchedulesRequest.workScheduleRequirements, delta)
+}
+
+/**
+ * Convert work schedules into timetable spacing requirements. This is not entirely semantically
+ * correct, but it lets us avoid work schedules like any other kind of time-bound constraint
```

Comment:
Why do you say "This is not entirely semantically correct"? I believe a work schedule is really similar to a spacing requirement (reserving some zone during a certain time). Is there something else you have in mind with your comment?

Reply:
They're very similar, but they're not truly spacing requirements, so it's technically a hack, albeit it is 99% justified.

Reply:
When talking with @Caracol3, he mentioned that when he was working on that kind of stuff, he really considered occupation of a track section by a train or by a work schedule the same way. So I would say the comment is not necessary.

Reply:
Works for me.